apps/rush-lib/src/logic/buildCache/AmazonS3BuildCacheProvider.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { Terminal } from '@rushstack/node-core-library';
+
+import { EnvironmentConfiguration, EnvironmentVariableNames } from '../../api/EnvironmentConfiguration';
+import { CloudBuildCacheProviderBase } from './CloudBuildCacheProviderBase';
+import { S3Client, GetObjectCommand, PutObjectCommand, GetObjectCommandOutput } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
```

Comment:
For a benchmark, I built 4 different branches of Rush and then invoked [Rundown](https://www.npmjs.com/package/@rushstack/rundown) with this command line:
```shell
$ npm install --global @rushstack/rundown
$ cd apps/rush-lib
$ rundown inspect --script ./lib/start.js
```

The results look like this:
- **Rush 5.33.1** loads **269** modules, right after @sachinjoseph had finished optimizing startup perf.<br/> [rundown-inspect-0-basline-5.33.1.log](https://github.com/microsoft/rushstack/files/5955309/rundown-inspect-0-basline-5.33.1.log)  
- **Rush 5.35.0** loads **299** modules, the last release before the cloud build cache feature was introduced.  <br/>[rundown-inspect-1-pre-cloud-5.35.0.log](https://github.com/microsoft/rushstack/files/5955305/rundown-inspect-1-pre-cloud-5.35.0.log)
- **Rush 5.38.0** loads **569** modules, the current `master` branch with @iclanton's Azure dependencies.<br/>[rundown-inspect-2-master-5.38.0.log](https://github.com/microsoft/rushstack/files/5955306/rundown-inspect-2-master-5.38.0.log)
- **This PR branch** loads **864** modules, with the new AWS dependencies. <br/>[rundown-inspect-3-aws-branch.log](https://github.com/microsoft/rushstack/files/5955308/rundown-inspect-3-aws-branch.log)

That's a significant regression over time.  But we should be able to get it under control relatively easily with a few judiciously placed `Import.lazy()` calls.

(Of course what's really missing here is actively measuring our perf. The Rundown tool is designed to automate that by generating snapshot files that can be diffed by Git; we just haven't hooked it up yet.)

Reply:
👍👍 Nice! With your latest commit, we're down to **349** modules.

[rundown-inspect.log](https://github.com/microsoft/rushstack/files/5969593/rundown-inspect.log)

Reply:
After the latest changes, we're at 358.

[rundown-inspect.log](https://github.com/microsoft/rushstack/files/6113533/rundown-inspect.log)